### PR TITLE
Correct the error in test code, fix bug in page.tsx

### PR DESCRIPTION
**Pull request by kexinF**

### CODE REVIEW

File: app/main/page.tsx
Comment (blickly) [inline app/main/page.tsx:1]: I don't understand the changes to this file.

File: app/main/NameTag.tsx [hunk 1]
```diff
@@ -101,11 +101,14 @@ function NameTag({
...
-          checked={showNametag}
-          onChange={() => setShowNametag(!showNametag)}
+          inputProps={{
+            role: 'switch',
+            'aria-checked': showNametag,
+            checked: showNametag,
+            value: showNametag.toString(),
+            onChange: () => setShowNametag(!showNametag)
+          }}
+          checked={showNametag} // It ensures the switch visually changes when clicked
```
Comment (blickly) [inline app/main/NameTag.tsx:111]: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

It looks like the checked state is set 4 different times in the code.  I think it should only be set at most once (doesn't a checkbox already keep track of it's own state)?

File: app/main/NameTag.tsx [hunk 2]
```diff
@@ -101,11 +101,14 @@ function NameTag({
       <div>
         <h2 style={{ fontSize: '30px', fontWeight: 'bold', display: 'inline-block' }}>Name Tag</h2>
         <ColoredSwitch
-          role='switch'
-          data-testid = 'colored-switch'
-          aria-checked={showNametag}
-          checked={showNametag}
-          onChange={() => setShowNametag(!showNametag)}
+          inputProps={{
+            role: 'switch',
```
Comment (blickly) [inline app/main/NameTag.tsx:105]: According to https://mui.com/material-ui/react-switch/#accessibility, we should test that the assistive technology our users will use supports the `switch` role properly before setting this. Have we done that?

File: __tests__/nametag.test.tsx [hunk 3]
```diff
@@ -2,32 +2,52 @@ import '@testing-library/jest-dom'
...
+        setCurrentNameTag={setCurrentNameTag}
+        setNameTagStatus={setNameTagStatus}
+        selectedWaveHand={selectedWaveHand}
+        waveHands={waveHands}
+      />
+    );
+
+    let switchInput = screen.getByRole('switch')
     expect(switchInput).not.toBeChecked();
-
```
Comment (blickly) [inline __tests__/nametag.test.tsx:26]: According to https://mui.com/material-ui/react-switch/#accessibility, switches/checkboxes should always have associated labels.  Can we make the test check that as well?

**Review by blickly — CHANGES_REQUESTED** [review]

File: __tests__/nametag.test.tsx [hunk 4]
```diff
@@ -9,25 +9,24 @@ jest.mock('../lib/zoomapi', () => jest.requireActual('../lib/fakezoomapi'));
...
-    render(<Page />)
 
-    let switchInput = screen.getByTestId('colored-switch');
-    expect(switchInput).not.toBeChecked();
+    render(<Page />)
 
-    // fireEvent.change(switchInput, { target: { 'aria-checked': true } });
-    await userEvent.click(switchInput);
+    const element = screen.getByLabelText('Name Tag');
+    expect(element).toBeInTheDocument();
```
Comment (blickly) [inline __tests__/nametag.test.tsx:25]: Can we check more specifically that the switch input and the element labeled "Name Tag" are the same component?

File: app/main/page.tsx
Comment (kexinF) [reply app/main/page.tsx:1]: solved, reverted the changes and will be do that in next PR

**Review by blickly — APPROVED** [review]
Looks good

**Pull request merged**